src/main/java/seedu/address/logic/parser/student/FilterStudentCommandParser.java

```diff
@@ -41,41 +41,36 @@ public FilterStudentCommand parse(String args) throws ParseException {
         boolean isTagsOnwards = false;
         List<String> listToReturn = new ArrayList<>();
         for (int i = 0; i < keywords.length; i++) {
-            if (keywords[i].startsWith("t/")) {
-                if (!keywords[i].substring(2).matches(TAG_REGEX)) {
-                    throw new ParseException(
-                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterStudentCommand.MESSAGE_USAGE));
-                }
+            isTagsOnwards = processEachWord(keywords, isTagsOnwards, listToReturn, i);
+        }
+        return listToReturn;
+    }
+
+    private boolean processEachWord(String[] keywords, boolean isTagsOnwards, List<String> listToReturn, int i)
+            throws ParseException {
+        if (!isTagsOnwards) {
+            if (keywords[i].startsWith("t/") && isValidFirstTag(keywords[i])) {
+                isTagsOnwards = true;
+                listToReturn.add(keywords[i]);
```

Comment:
Alright, I wish the code could be neater as well, but it seems like there's nothing much we can do about it at this late stage. Could you perhaps add in some javadocs before I merge it? Ie. maybe give a toy example of how a filterStudent command gets passed down, and how parameters are parsed and passed around.

Reply:
done